modules/dists/BlockDist.chpl

```diff
@@ -44,8 +44,8 @@ private use CommDiagnostics;
 private use ChapelLocks;
 private use ChapelDebugPrint;
 
-use SparseBlockDist;
-use LayoutCS;
+public use SparseBlockDist;
+public use LayoutCS;
```

Comment:
Did you run into any non-trivial issues if these were private? (Or default, after this PR)

I am mostly concerned about `LayoutCS` being used indirectly if a user `use`s `BlockDist`. Which is probably the case before this PR, but it'd be nice to prevent that with this PR, maybe? Arguably, this and other potential similar cases can be follow up to this, as well.

Reply:
I didn't change anything that didn't require changing by at least one test, but that's not to say that I always made the best decisions.  My memory is that any test that uses sparse subdomains of block distributions requires SpareBlockDist, and that doesn't particularly surprise me.  Or at least, I'm not sure what to do about it or even whether we could do anything about it at this point (due to issue #14407).

I think that some of the LayoutCS uses were also related to those tests, and when I found the `use` here, I admittedly assumed that this was a reasonable place for it without thinking about it very hard (but thinking about it now, I agree that it seems more regrettable).

I could re-run without that `use` to see what breaks and whether it would be more appropriate to push that `use` into the test itself.  (Or you could :)...  as you point out, I didn't actually change the behavior so much as make it more obvious and exposed).  But I agree that it'd be nice to get these as locked down as we think they ought to be (and can be), either here, or in a follow-on PR.

Reply:
I think my concern and this PR's concern are logically separate enough. So, no need for additional effort in this PR. I was just curious whether there is any major obstacle for doing that. I don't mind taking a look and see if the change (particularly for `LayoutCS`) is easy enough after this PR.

Reply:
Here's a list of the tests that failed if LayoutCS was not publicly used.  I think my quick analysis is/was that its use as a default argument requires it to be visible at the callsite of a sparse block domain's creation point.  But I don't feel completely confident in that assessment (and it may not apply to all tests):

```
[Error matching compiler output for distributions/bradc/spsBlk/blockSparse2CSR]
[Error matching compiler output for sparse/engin/dsiFuncsSpsBlk]
[Error matching compiler output for sparse/engin/nnzCounting]
[Error matching compiler output for studies/bale/toposort/toposort]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTest]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestCSR]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestDist]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestDistBlockCyc]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestDistCyc]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestDistSpsBlk]
[Error matching compiler output for users/engin/partial_reduction_support/template_tests/templateTestSparse]
[Error matching compiler output for users/engin/partial_reduction_support/unit_tests/densePartialThese]
[Error matching compiler output for users/engin/partial_reduction_support/unit_tests/partialDomainDefaultRectangular]
[Error matching compiler output for users/engin/partial_reduction_support/unit_tests/sparsePartialThese (compopts: 1)]
[Error matching compiler output for users/engin/partial_reduction_support/unit_tests/sparsePartialThese (compopts: 2)]
[Error matching compiler output for users/engin/partial_reduction_support/unit_tests/tupleMethods]
[Error matching compiler output for users/engin/sparse_bulk_dist/integrity (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/integrity (compopts: 2)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseAdd2D (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseAdd2D (compopts: 2)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulk (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulk (compopts: 2)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulk2D (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulk2D (compopts: 2)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulkBoundsCheck (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulkBoundsCheck (compopts: 2)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulkOpts (compopts: 1)]
[Error matching compiler output for users/engin/sparse_bulk_dist/sparseBulkOpts (compopts: 2)]
[Error matching compiler output for users/engin/sparse_index_buffer/sparseIndexBuffer]
```